File: selectionSort/selectionSort.cpp
#include <iostream>

int* selectionSort(int arr[], int length) {
	for (unsigned int startIndex = 0; startIndex < length - 1; ++startIndex) {
		int smallestIndex = startIndex;

		for (unsigned int currentIndex = startIndex + 1; currentIndex < length; ++currentIndex) {
			if (arr[currentIndex] < arr[smallestIndex]) {
				smallestIndex = currentIndex;
			}
		}

		std::swap(arr[startIndex], arr[smallestIndex]);
	}

	return arr;
}

int main(void) {
	int arr[5] = { 4, 1, 7, 8, 9 };
	int length = sizeof(arr) / sizeof(arr[0]);

	int* sortedArr = selectionSort(arr, length);

	for (unsigned int index = 0; index < length; ++index) {
		std::cout << sortedArr[index] << "\t";
	}

	return 0;
}

File: javaScript/selectionSort/selectionSort.js
let selectionSort = (array) => {
    for (let startIndex = 0; startIndex < array.length - 1; startIndex++) {
        let smallestIndex = startIndex;

        for (let currentIndex = startIndex + 1; currentIndex < array.length; currentIndex++) {
            if (array[smallestIndex] > array[currentIndex]) {
                smallestIndex = currentIndex;
            }
        }

        [array[startIndex], array[smallestIndex]] = [array[smallestIndex], array[startIndex]];
    }

    return array;
}

let array = [1, 4, 2, 8, 345, 123, 43, 32, 5643, 63, 123, 43, 2, 55, 1, 234, 92];
console.log(selectionSort(array));